sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -1980,3 +2089,23 @@ class ExistsThrowsExceptionFileSystem extends RawLocalFileSystem {
 object ExistsThrowsExceptionFileSystem {
   val scheme = s"FileStreamSourceSuite${math.abs(Random.nextInt)}fs"
 }
+
+class CountListingLocalFileSystem extends RawLocalFileSystem {
```

Comment:
The code regarding FileSystem I add here is very similar with what I add in #27664. When either one gets merged, I'll rebase and deduplicate it.

Reply:
Could you close this comment?